manual/io/index.md

```diff
@@ -56,784 +46,433 @@ root [1] TFile f1("my.root");
 root [2] gDirectory->pwd()
 
 my.root:/
+root [3] hist->Draw() // for `hist` from `my.root`
 {% endhighlight %}
 
 `Rint:/` corresponds to "ROOT's in-memory" directory, the default directory during startup.
 
-## Viewing the contents of a ROOT file
-
-{% include ref class="TFile" %} is a descendent of {% include ref class="TDirectory" %}, which means it behaves like a `TDirectory`. You can list the contents, print the
-name, and create subdirectories. In a ROOT session, you are always in a directory and the directory you are in is
-called the current directory and is stored in the global variable `gDirectory` (see → [gDirectory]({{ '/manual/root_architecture_and_components/#gdirectory' | relative_url }})).
+### Adding and removing objects from a ROOT file
 
-### Physical layout of a ROOT file
+When writing an object to a ROOT file, ROOT creates a directory "entry" ({% include ref class="TKey" %}) representing the object, which mainly consists of a name and the object's persistent data.
+You can think of a `TFile` as a collection of `TKeys`, possibly inside nested `TDirectory`-s.
 
-- Use the [TFile::Map()](https://root.cern/doc/master/classTFile.html#a5568f2f0a4a678ffaf769d0bf210610f){:target="_blank"} method to view the physical layout of a ROOT file.
-
-The output prints the date/time, the start record address, the number of bytes in the record, the class name of the record and the compression factor.
+The name of the `TKey` can be either explicitly stated when writing, or it can be determined from `TObject::GetName()` for classes inheriting from `TObject`.
 
 _**Example**_
 
 {% highlight C++ %}
-root[] f.Map()
-
-20191010/122600    At:100     N=114    TFile
-20191010/122600    At:214     N=429    TH1F           CX = 2.31
-20191010/122600    At:643     N=424    TH1F           CX = 2.33
-20191010/122600    At:1067    N=426    TH1F           CX = 2.32
-20191010/122600    At:1493    N=425    TH1F           CX = 2.33
-20191010/122600    At:1918    N=429    TH1F           CX = 2.31
-20191010/122600    At:2347    N=424    TH1F           CX = 2.33
-20191010/122600    At:2771    N=418    TH1F           CX = 2.37
-20191010/122600    At:3189    N=428    TH1F           CX = 2.31
-20191010/122600    At:3617    N=422    TH1F           CX = 2.34
-20191010/122600    At:4039    N=421    TH1F           CX = 2.35
-20191010/122600    At:4460    N=431    TH1F           CX = 2.30
-20191010/122600    At:4891    N=424    TH1F           CX = 2.34
-20191010/122600    At:5315    N=430    TH1F           CX = 2.31
-20191010/122600    At:5745    N=426    TH1F           CX = 2.33
-20191010/122600    At:6171    N=425    TH1F           CX = 2.34
-20191010/122600    At:6596    N=3055   StreamerInfo   CX = 3.08
-20191010/122600    At:9651    N=732    KeysList
-20191010/122600    At:10383   N=53     FreeSegments
-20191010/122600    At:10436   N=1      END
+void WriteHist(TH1* hist) {
+  gDirectory->WriteObject(hist, "explicitName");
+  // OR:
```

Comment:
```suggestion
  // OR (less preferred):
```

Reply:
I don't think so, why?

Reply:
For style reasons I believe, but it's not necessary